.github/workflows/trunk.yml

```diff
@@ -55,6 +55,31 @@ jobs:
         # Build and test executorch
         PYTHON_EXECUTABLE=python ${CONDA_RUN} bash .ci/scripts/test_model.sh "${MODEL_NAME}" "${BUILD_TOOL}" "${BACKEND}"
 
+  test-models-macos-coreml:
+    name: test-models-macos-coreml
+    uses: pytorch/test-infra/.github/workflows/macos_job.yml@main
+    strategy:
+      matrix: 
+        model: [dl3, edsr, efficient_sam, emformer_join, emformer_transcribe, ic3, mobilebert, mv3, resnet50, vit, w2l]
+        backend: [coreml]
+      fail-fast: false
+    with:
+      runner: macos-m1-stable
```

Comment:
IIRC this runner also had bandwidth issue and things would timeout. I am ok with this, let's see what happens.